### PR TITLE
feat: multiship payment

### PR DESCRIPTION
**Pull request by kibo-chandradeeptalaha**

### CODE REVIEW

File: components/checkout/PaymentStep/PaymentStep.tsx [hunk 1]
```diff
@@ -320,20 +326,35 @@ const PaymentStep = (props: PaymentStepProps) => {
 
     selectedCards?.forEach(async (card: Maybe<CrPayment>) => {
       paymentAction = { ...paymentAction, actionName: 'VoidPayment' }
-      await updateOrderPaymentAction.mutateAsync({
-        orderId: checkout?.id as string,
-        paymentId: card?.id as string,
-        paymentAction,
-      })
+      if (isMultiShip) {
```
Comment (kibo-kevinwatts) [inline components/checkout/PaymentStep/PaymentStep.tsx]: After looking at this, we could maybe modify it to perform via props for this void action and the add action

```await onVoidPayment({ id: checkout?.id as string, paymentId: card?.id as string, paymentAction }) ```
and 
```await onAddPayment({ id: checkout.id, paymentAction, billingInfoInput })```

The `onAddPayment` method can be defined in the parent template and handle the `if(multiship) logic`

Comment (kibo-chandradeeptalaha) [reply components/checkout/PaymentStep/PaymentStep.tsx]: Done.

File: components/checkout/PaymentStep/PaymentStep.tsx [hunk 2]
```diff
@@ -44,15 +37,17 @@ import type {
...
+  Checkout,
 } from '@/lib/gql/types'
 
 interface PaymentStepProps {
-  checkout: CrOrder | undefined
+  checkout: CrOrder | Checkout
   contact?: ContactForm
+  isMultiShip?: boolean
+  onVoidPayment: (id: string, paymentId: string, paymentAction: PaymentActionInput) => void
+  onAddPayment: (id: string, paymentAction: PaymentActionInput) => void
```
Comment (kibo-sushant) [inline components/checkout/PaymentStep/PaymentStep.tsx:50]: Verify through Unit Testcases whether 
 onVoidPayment and onAddPayment are getting called or not

File: components/page-templates/MultiShipCheckoutTemplate/MultiShipCheckoutTemplate.tsx [hunk 3]
```diff
@@ -150,9 +176,12 @@ const MultiShipCheckoutTemplate = (props: CheckoutProps) => {
...
         )}
-        {/* @to-do Use below steps for future development */}
-        {/* <PaymentStep checkout={checkout} {...paymentStepParams} />
-              <ReviewStep checkout={checkout as Checkout} onBackButtonClick={handleBack} /> */}
+        <PaymentStep
+          checkout={checkout as Checkout}
+          onVoidPayment={handleVoidPayment}
+          onAddPayment={handleAddPayment}
+        />
+        {/* <ReviewStep checkout={checkout as Checkout} onBackButtonClick={handleBack} /> */}
```
Comment (kibo-sushant) [inline components/page-templates/MultiShipCheckoutTemplate/MultiShipCheckoutTemplate.tsx:184]: Unit Testcases are missing

File: components/page-templates/StandardShipCheckoutTemplate/StandardShipCheckoutTemplate.tsx [hunk 4]
```diff
@@ -88,6 +91,32 @@ const StandardShipCheckoutTemplate = (props: CheckoutProps) => {
...
+  }
+
+  const handleAddPayment = async (id: string, paymentAction: PaymentActionInput) => {
+    await createOrderPaymentMethod.mutateAsync({ orderId: id, paymentAction })
+    await updateCheckoutBillingInfo.mutateAsync({
+      orderId: id,
+      billingInfoInput: { ...paymentAction.newBillingInfo },
+    })
+  }
+
```
Comment (kibo-sushant) [inline components/page-templates/StandardShipCheckoutTemplate/StandardShipCheckoutTemplate.tsx:119]: Unit Testcases are missing

File: hooks/mutations/multiShip/useUpdateCheckoutPaymentActionMutation/useUpdateCheckoutPaymentActionMutation.ts [hunk 5]
```diff
@@ -0,0 +1,37 @@
...
+
+  return useMutation(updateCheckoutPayment, {
+    onSuccess: () => {
+      queryClient.invalidateQueries(checkoutKeys.all)
+    },
+  })
+}
+
+export const useUpdateMultiShipCheckoutPaymentActionMutation =
+  useUpdateCheckoutPaymentActionMutation
```
Comment (kibo-sushant) [inline hooks/mutations/multiShip/useUpdateCheckoutPaymentActionMutation/useUpdateCheckoutPaymentActionMutation.ts:55]: TypeDoc code missing

File: components/page-templates/StandardShipCheckoutTemplate/StandardShipCheckoutTemplate.tsx (hunk 4 above)
Comment (kibo-chandradeeptalaha) [reply components/page-templates/StandardShipCheckoutTemplate/StandardShipCheckoutTemplate.tsx:119]: Handling in different PR

File: components/page-templates/MultiShipCheckoutTemplate/MultiShipCheckoutTemplate.tsx (hunk 3 above)
Comment (kibo-chandradeeptalaha) [reply components/page-templates/MultiShipCheckoutTemplate/MultiShipCheckoutTemplate.tsx:184]: Handling in different PR

File: components/checkout/PaymentStep/PaymentStep.tsx (hunk 2 above)
Comment (kibo-chandradeeptalaha) [reply components/checkout/PaymentStep/PaymentStep.tsx:50]: We need a separate ticket to handle this as there are many test cases remaining for this

File: hooks/mutations/multiShip/useUpdateCheckoutPaymentActionMutation/useUpdateCheckoutPaymentActionMutation.ts (hunk 5 above)
Comment (kibo-chandradeeptalaha) [reply hooks/mutations/multiShip/useUpdateCheckoutPaymentActionMutation/useUpdateCheckoutPaymentActionMutation.ts:55]: Done

**Pull request closed without merging**